docs/why-ngrok.mdx

```diff
@@ -0,0 +1,132 @@
+business logic is pushed to proxies in front of the application, the contract
+between the reverse proxy and the upstream application becomes critical.
+Unexpected changes in behavior between them cause incidents. Traditionally,
+that contract is _implicitly_ defined in IT tickets.
+
+ngrok's Agent SDKs change that. Instead, the application itself defines via API
+exactly what operations it expects the ingress platform to perform on its
+behalf and how it expects to receive information about those behaviors.
+
+This is a new, safer programming model where developers can safely offload
+additional functionality to the ingress platform without fear of the contract
+being broken.
+
+## Pay as you go
+
+Get started without a huge bill.
+
+Unlike other networking software which has unpublished pricing that only
+enterprises can afford, anyone can get started using the paid features of ngrok
+with just a credit card. ngrok's pricing is public, predictable and you pay
+only for what you use.
+
+## Dev / Prod Parity
+
+ngrok eliminates bugs from differences between local dev and production.
+
+We've all felt the pain of something failing in production that worked while we
+were testing in local. Because ngrok works everywhere, you can run it on your
+local development environment exactly the same way you run it to deliver your
+production service.
+
+## As-a-Service
+
+ngrok reduces your operational burden by operating the global delivery network
+for you.
+
+A global delivery network accelerates your application, but it's a real pain to
+operate. Unlike deployable network software that you need to configure and run
+yourself, ngrok is a completely managed global service. You can spend less time
+operating networking infrastructure and puzzling over missing packets and more
+time building value for your customers.
+
+## Built for Developers
+
+ngrok puts developers in control with best-in-class developer tooling.
+
+- First-class CLI to all functionality
+- Native-language Agent SDKs
+- HTTP API to automate configuration tasks
+- Exhaustive documentation
+- Programmatic events for observability and automation
+
+## Global Acceleration
+
+ngrok accelerates your apps for global clients.
+
+In addition to using its global server load balancing to improve speeds for
+your application's clients by reducing TCP and TLS setup round-trip times,
+ngrok also pushes any module behaviors you configure to its global edge as well
+so authentication, transformations, load balancing and more happen as close to
+your customers as possible.
+
+## Network Micro-Segmentation
+
+ngrok finally achieves true per-workload layer 4 networking security.
+
+Traditional networking architectures often permit application to speak to each
+other directly when they're colocated on the same network without going through
+the ingress gateways which enforce auth and policy. This is ultimately the same
+security risk that motivated beyond-corp zero-trust architectures. ngrok
```

Comment:
security risk that motivated Google to create its BeyondCorp zero-trust architecture model. ngrok